crates/oxc_allocator/src/clone_in.rs

```diff
@@ -31,6 +31,12 @@ pub trait CloneIn<'new_alloc>: Sized {
     /// Clone `self` into the given `allocator`. `allocator` may be the same one
     /// that `self` is already in.
     fn clone_in(&self, allocator: &'new_alloc Allocator) -> Self::Cloned;
+
+    /// Almost same as `clone_in`, but for some special type, it will also clone the semantic ids.
+    #[inline]
+    fn clone_in_with_semantic_ids(&self, allocator: &'new_alloc Allocator) -> Self::Cloned {
+        self.clone_in(allocator)
+    }
```

Comment:
Please expand this comment to explain why it's generally a bad idea to use this method.